src/solvers/dg/2d/dg.jl

```diff
@@ -21,13 +21,16 @@ mutable struct Dg2D{Eqn<:AbstractEquation, NVARS, POLYDEG,
 
   boundaries::BoundaryContainer2D{NVARS, POLYDEG}
   n_boundaries::Int
+  n_boundaries_per_direction::SVector{4, Int}
 
   mortar_type::MortarType
   l2mortars::L2MortarContainer2D{NVARS, POLYDEG}
   n_l2mortars::Int
   ecmortars::EcMortarContainer2D{NVARS, POLYDEG}
   n_ecmortars::Int
 
+  boundary_conditions
```

Comment:
Well, since all boundary conditions are their own function, the alternative would be something like `::NTuple{4, Any}`, wouldn't it? I could also add a dummy "no-op" BC that replaces `nothing`, but then we would still end up with the non-concrete type `::NTuple{4, Function}`, wouldn't we?

I was aware of the issue but thought that this will sort itself out once Taal becomes alive. But if you have a suggestion for the meantime, I"d be more than happy to apply it.

Reply:
> I was aware of the issue but thought that this will sort itself out once Taal becomes alive.

That's okay for me. Alternatively, you could introduce and additional type parameter of `Dg2D` which is `BoundaryConditions<:NTuple{4,Any}` and use `boundary_conditions::BoundaryConditions`.

Reply:
What would be the benefit of this? Isn't this still type-unstable?

Reply:
`boundary_conditions::BoundaryConditions` is specialized and `BoundaryConditions<:NTuple{4,Any}` makes sure that you have 4 BCs. That's fully typed, but `boundary_conditions[direction]` below will still be type-unstable.

Reply:
I did something similar in 51028d2. Is this as good as what you had in mind or did I make a mistake?

Reply:
Looks good to me. Although I'm not sure whether it's needed after introducing the function barrier.